internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -507,49 +515,41 @@ func monitorOperands(ctx context.Context, log logrus.FieldLogger, asc ASC) error
 	for _, deployName := range []string{"agentinstalladmission", "assisted-service"} {
 		deployment := &appsv1.Deployment{}
 		if err := asc.Client.Get(ctx, types.NamespacedName{Name: deployName, Namespace: asc.namespace}, deployment); err != nil {
-			return err
+			return "", err
 		}
 
 		if isStatusConditionFalse(deployment.Status.Conditions, appsv1.DeploymentAvailable) {
 			msg := fmt.Sprintf("Deployment %s is not available", deployName)
 			log.Error(msg)
-			return pkgerror.New(msg)
+			return msg, nil
 		}
 
 		if isStatusConditionFalse(deployment.Status.Conditions, appsv1.DeploymentProgressing) {
 			msg := fmt.Sprintf("Deployment %s is not progressing", deployName)
 			log.Error(msg)
-			return pkgerror.New(msg)
+			return msg, nil
 		}
 	}
 
 	// monitor statefulset
 	ss := &appsv1.StatefulSet{}
 	if err := asc.Client.Get(ctx, types.NamespacedName{Name: imageServiceName, Namespace: asc.namespace}, ss); err != nil {
-		return err
+		return "", err
 	}
 
 	desiredReplicas := *ss.Spec.Replicas
-	checkReplicas := func(replicas int32, name string) error {
+	for kind, replicas := range map[string]int32{
+		"created": ss.Status.Replicas,
+		"ready":   ss.Status.ReadyReplicas,
+		"current": ss.Status.CurrentReplicas,
+		"updated": ss.Status.UpdatedReplicas,
+	} {
 		if replicas != desiredReplicas {
-			return fmt.Errorf("StatefulSet %s %s replicas does not match desired replicas", imageServiceName, name)
+			return fmt.Sprintf("StatefulSet %s %s replicas does not match desired replicas", imageServiceName, kind), nil
```

Comment:
Should this return an error to indicate a monitoring failure?

Reply:
This isn't really a "monitoring failure" this is a situation where we successfully monitored the stateful set and determined it wasn't in a good state so I'm reporting that there's an issue, but not returning an error from the function.

Reply:
Gotcha, thanks for the explanation!